src/main.js

```diff
@@ -0,0 +1,171 @@
+const selectors = {
+    HOLE: '.hole',
+    GAME: '.game',
+    START_BTN: '.start',
+    SCORE:'.score',
+    MODAL: '.modal',
+    UP: '.up',
+    HOLE_7: '.hole7',
+    HOLE_8: '.hole8',
+};
+
+const classes = {
+    MOLE: 'mole',
+    UP: 'up',
+    ACTIVE: 'active',
+    CLOSE: 'close',
+    REPEAT_BTN: 'repeat-level-button',
+    NEXT_LEVEL_BTN: 'next-level-button',
+    NEW_GAME_BTN: 'new-game-button',
+    NEXT_LEVEL: 'next-level',
+};
+
+const holes = document.querySelectorAll(selectors.HOLE);
+const scoreBoardElement = document.querySelector(selectors.SCORE);
+const modalWindowElement = document.querySelector(selectors.MODAL);
+const hole7Element = document.querySelector(selectors.HOLE_7);
+const hole8Element = document.querySelector(selectors.HOLE_8);
+let score = 0;
+let minMoleDisappearanceRate;
+let maxMoleDisappearanceRate;
+let timerId;
+let lastHole;
+let isGameFinished = false;
+
+function getRandomTime(min, max) {
+    return Math.round(Math.random() * (max - min) + min);
+}
+
+function getRandomHole(holes) {    
+    const holesRandomElementIndex = Math.floor(Math.random() * holes.length);
+    const newHole = holes[holesRandomElementIndex];    
+
+    if (newHole === lastHole) {
+        return getRandomHole(holes);
+    }
+    lastHole = newHole;
+    return newHole;  
+}
+
+function peepMole() {
+    const time = getRandomTime(minMoleDisappearanceRate, maxMoleDisappearanceRate);
+    const hole = getRandomHole(holes);
+
+    if(!isGameFinished) {
+        hole.classList.add(classes.UP);
+        timerId = setTimeout(() => {
+            hole.classList.remove(classes.UP);
+            peepMole();
+        }, time);         
+    }        
+}
+
+function finishGame() {
+    isGameFinished = true;
+    document.querySelectorAll(selectors.UP).forEach((el) => el.classList.remove(classes.UP));
+    localStorage.setItem('score', score);
+    localStorage.setItem('min', minMoleDisappearanceRate);
+    localStorage.setItem('max', maxMoleDisappearanceRate);
```

Comment:
Let's move to a separate function - saveLevelDataToLocalStorage or smth like this